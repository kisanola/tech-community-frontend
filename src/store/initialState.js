module.exports = {
  currentUser: {
    isAuth: localStorage.getItem('token') !== null,
    user: { username: null },
  },
  feed: {
    items: [],
  },
  search: {
    searchKeywords: '',
    searchItems: [],
    searchState: {
      query: '',
    },
  },
  posts: {},
  sharePost: {
    show: false,
    content: {
      url: '',
      title: '',
      content: '',
    },
  },
  singlePost: {
    post: {
      author: {},
      slug: '',
      userType: '',
      description: '',
      likesCount: 0,
      createdAt: '',
    },
    relatedPosts: {
      isEmpty: false,
      feed: [],
    },
  },
  socialAuth: { show: false },
};
