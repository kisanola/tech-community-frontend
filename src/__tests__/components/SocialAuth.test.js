import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import SocialAuth from '../../components/Login/socialAuth';
import initialState from '../../store/initialState';

let wrapper;
const props = {
  handleShowAndHide: jest.fn(),
  show: true,
};

const mockStore = configureMockStore([thunk]);

describe('Header.jsx', () => {
  let store;
  beforeEach(() => {
    const socialAuth = { show: false };
    store = mockStore({
      ...initialState, socialAuth,
    });
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <SocialAuth {...props} />
        </Router>
      </Provider>,
    );
  });
  test('should render SocialAuth.jx', () => {
    // wrapper = shallow(<SocialAuth {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when click on `login with google/github` button', () => {
    beforeEach(() => {
      delete window.location;
      window.location = { replace: jest.fn() };
    });
    test('should hit the login with google endpoint', () => {
      wrapper.find('button.social-login-google').simulate('click');
      expect(window.location.replace).toHaveBeenCalled();
    });

    test('should hit the login with github endpoint', () => {
      wrapper.find('button.social-login-github').simulate('click');
      expect(window.location.replace).toHaveBeenCalled();
    });
  });
});
