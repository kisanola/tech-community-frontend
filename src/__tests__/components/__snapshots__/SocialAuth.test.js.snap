// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.jsx should render SocialAuth.jx 1`] = `ReactWrapper {}`;
